pkg/segment/writer/segstore.go

```diff
@@ -282,6 +296,11 @@ func (segstore *SegStore) resetSegStore(streamid string, virtualTableName string
 	segstore.wipBlock.colWips = make(map[string]*ColWip)
 	segstore.wipBlock.clearPQMatchInfo()
 
+	if segstore.SegmentErrors != nil {
+		segstore.LogSegmentErrors()
+		segstore.SegmentErrors = nil
+	}
```

Comment:
nit: If we don't have to log errors multiple times, it might make sense to rename `LogSegmentErrors()` to `FlushSegmentErrors()` or `LogAndFlushSegmentErrors()`—actually, you probably don't need "Segment" because it's a method on a segstore, which is already only for one segment—and then just call that function here.

This might just be my personal preference, but I also think you don't need this check `segstore.SegmentErrors != nil` because it's easier for the reader if they don't need to know that `LogSegmentErrors()` uses `SegmentErrors`; `LogSegmentErrors()` should just handle that case (e.g., do nothing and don't panic)

Reply:
addressed in the latest commit.